src/output.rs

```diff
@@ -392,14 +290,188 @@ impl SelectResource {
 }
 
 
+//------------ Output --------------------------------------------------------
+
+#[derive(Clone, Debug)]
+pub struct Output {
+    /// Limiting data to be included.
+    ///
+    /// If this is `None`, all data is potentially included.
+    selection: Option<Selection>,
+
+    /// Should we include route origins?
+    route_origins: bool,
+
+    /// Should we include router keys?
+    router_keys: bool,
+
+    /// Should we include ASPA data?
+    aspas: bool,
+}
+
+impl Output {
+    /// Creates new default output
+    pub fn new() -> Self {
+        Self {
+            selection: None,
+            route_origins: true,
+            router_keys: true,
+            aspas: true
+        }
+    }
+
+    /// Creates a new output based on the config.
+    pub fn from_config(config: &Config) -> Self {
+        let mut res = Self::new();
+        res.update_from_config(config);
+        res
+    }
+
+    pub fn from_query(query: Option<&str>) -> Result<Self, QueryError> {
+        let mut res = Self::new();
+        res.update_from_query(query)?;
+        Ok(res)
+    }
+
+    pub fn update_from_config(&mut self, config: &Config) {
+        if !config.enable_bgpsec {
+            self.no_router_keys();
+        }
+        if !config.enable_aspa {
+            self.no_aspas();
+        }
+    }
+
+    /// Updates the output value from query parameters.
+    pub fn update_from_query(
+        &mut self, query: Option<&str>
+    ) -> Result<(), QueryError> {
+        let query = match query {
+            Some(query) => query,
+            None => return Ok(())
+        };
+
+        let mut selection = Selection::new();
+        for (key, value) in form_urlencoded::parse(query.as_ref()) {
+            if key == "select-prefix" || key == "filter-prefix" {
+                selection.resources.push(
+                    SelectResource::Prefix(addr::Prefix::from_str(&value)?)
+                );
+            }
+            else if key == "select-asn" || key == "filter-asn" {
+                selection.resources.push(
+                    SelectResource::Asn(
+                        Asn::from_str(&value).map_err(|_| QueryError)?
+                    )
+                );
+            }
+            else if key == "include" {
+                for value in value.split(',') {
+                    #[allow(clippy::single_match)]
+                    match value {
+                        "more-specifics" => selection.more_specifics = true,
+                        _ => { }
+                    }
+                }
+            }
+            else if key == "exclude" {
+                for value in value.split(',') {
+                    match value {
+                        "origins" => self.route_origins = false,
```

Comment:
This is inconsistent with the command line switch `--no-route-origins`.

Reply:
I was about to argue, but yeah, you are right.